src/howitz/endpoints.py

```diff
@@ -142,27 +124,55 @@ def get_event_details(id):
 
 @app.route('/')
 @app.route('/hello-world')
+@login_required
 def index():
     exemplify_loop = list('abracadabra')
     return render_template('index.html', example_list=exemplify_loop)
 
 
 @app.route('/events')
+@login_required
 def events():
     # current_app["expanded_events"] = []
     return render_template('/views/events.html')
 
 @app.route('/login')
 def login():
-    return render_template('/views/login.html')
+    print("current user", current_user.is_authenticated)
+    if current_user.is_authenticated:
+        default_url = flask.url_for('index')
+        # print("DEFAULT URL", default_url)
+        return flask.redirect(default_url)
+    else:
+        return render_template('/views/login.html')
 
 @app.route('/sign_in_form')
 def sign_in_form():
     return render_template('/components/login/sign-in-form.html')
 
 @app.route('/auth', methods=["POST"])
 def auth():
-    return render_template('/views/events.html')
+    username = request.form["username"]
+    password = request.form["password"]
+
+    user = authenticate_user(username, password)
+    if user:
+        print("User", user)
+        login_user(user)
+        flask.flash('Logged in successfully.')
+
+        # redirect to /events
+        resp = make_response()
+        resp.headers['HX-Redirect'] = '/events'
+        return resp
+    else:
+        pass
```

Comment:
You don't need the `pass` here since there is more stuff on the same indent-level.